ciao-scheduler/scheduler.go

```diff
@@ -183,6 +204,7 @@ func connectController(sched *ssntpSchedulerServer, uuid string) {
 
 	if sched.controllerMap[uuid] != nil {
 		glog.Warningf("Unexpected reconnect from controller %s\n", uuid)
+		sched.controllerMutex.Unlock()
```

Comment:
Won't this cause a panic?  When the function exits the mutex will be unlocked for the second time by the deferred function and we'll get a runtime error.